benchmark/reformat_input_gi.py

```diff
@@ -0,0 +1,143 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, The WGS-HGT Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+"""
+Reformat input genome files to formats required by given
+GI tool
+=========================================================
+"""
+
+import click
+from os import remove
+from os.path import join
+
+from skbio import Sequence, DNA
+
+
+def reformat_egid(genbank_fp,
+                  output_dir):
+    """ Reformat input genome to the formats accepted by EGID.
+
+    Parameters
+    ----------
+    genbank_fp: string
+        file path to genome in GenBank format
+    output_dir: string
+        output directory path
+
+    Notes
+    -----
+    Input to EGID are five obsolete NCBI standard files: gbk, fna, faa, ffn
+    and ptt.
+    """
+    loci = []
+    nucl_seq = ''
+    genes = {}
+    nseq = 0  # number of nucleotide sequences
+    with open(genbank_fp, 'r') as input_f:
+        for line in input_f:
+            if line.startswith('//'):
+                nseq += 1
+    abs_pos = 0  # absolute position in concantenated nucleotide sequence
+    for i in range(nseq):
+        gb = Sequence.read(genbank_fp, seq_num=i+1, format='genbank')
+        locus_name = gb.metadata['LOCUS']['locus_name']
+        size = gb.metadata['LOCUS']['size']
+        loci.append([locus_name, size])
+        nucl_seq += str(gb)
+        for feature in gb.interval_metadata.features:
```

Comment:
When I wrote these codes I was inspired by some previous [codes](https://github.com/biocore/WGS-HGT/blob/master/benchmark/simulate_hgts.py#L47). Therefore you may find that the style is similar. Here is s question if we make this a function: a global variable `genes` is outside the loop. If it is an independent function, it will have to access `genes` every iteration. I guess that it may be complicated. I am not very familiar with Python's namespace and memory allocation policies. So please correct me if I am not getting this right.